virtual/registry/kv_registry.go

```diff
@@ -258,9 +258,11 @@ func (k *kvRegistry) EnsureActivation(
 			timeSinceLastHeartbeat           = time.Since(server.LastHeartbeatedAt)
 			serverID                         string
 			serverAddress                    string
+			serverVersion                    int64
 		)
 		if activationExists && serverExists && timeSinceLastHeartbeat < HeartbeatTTL {
 			// We have an existing activation and the server is still alive, so just use that.
+			serverVersion = server.ServerVersion
```

Comment:
I thought you would have to associated the server's version with the current activation instead of looking it up on the server heartbeat key. I thought about it for a long time though and couldn't come up with any reason why this is incorrect. My intuition for why this is fine is as follows, maybe add a comment along these lines?

"Its ok to lookup the ServerVersion from the server discovery key directly because if the activation is still active, then by definition the server's version has not changed since the activation was first created."

Let me know if that makes sense